### PR TITLE
fix(filter-field, autocomplete): Fixes an issue that the panels did n…

### LINKED ISSUE #1747: Filter Field Autocomplete Panel appears above page header
# Bug Report

## Expected Behavior

autocomplete panel should not overlay sticky header 

## Current Behavior

autocomplete panel overlays sticky header

## Steps to Reproduce

https://stackblitz.com/edit/github-denej4?file=src/app/app.component.html

1. Expand the panel by click in `Filter by` input
2. With panel expanded, scroll up

## Context (Environment)

I can see in Stackblitz that the `dt-autocomplete-panel` is rendered inside `cdk-overlay-container`, `cdk-overlay-container` is rendered outside the app root and has z-index: 1000 thus covering everything. 

**Exactly same issue was reported and fixed in https://github.com/dynatrace-oss/barista/issues/451.** 

Is Stackblitz example valid and something might have broken in Barista after https://github.com/dynatrace-oss/barista/issues/451 fix? 
Do I miss something in the Stackblitz example that would make the `dt-autocomplete-panel` overlay respect fixed header positioning? 

### Used Versions:

- **angular**: 10.1.6
- **@angular/cdk**: 10.2.5
- **@dynatrace/barista-components**: 8.1.2

### PR DESCRIPTION
**Pull request by ffriedl89**
…ot react to viewport boundaries correctly.

Closes #1747.

### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by lukasholzer — CHANGES_REQUESTED** [review]
THx for adding a test :) Good idea to put it in a mixin 👍 

Can you please use the mixin in the selection area as well?

**Review by lukasholzer — DISMISSED** [review]
Thx for changing in the selection area as well.

File: libs/barista-components/chart/src/selection-area/selection-area.ts [hunk 1]
```diff
@@ -189,18 +193,12 @@ export class DtChartSelectionArea implements AfterContentInit, OnDestroy {
     private _changeDetectorRef: ChangeDetectorRef,
     // tslint:disable-next-line: no-any
     @Inject(DOCUMENT) private _document: any,
-    @Optional() private _viewportResizer: DtViewportResizer,
-  ) {}
+    @Optional() public _viewportResizer: DtViewportResizer,
```
Comment (tomheller) [inline libs/barista-components/chart/src/selection-area/selection-area.ts:196]: Do we want this in the breaking-changes list, as we do in the filter-field range, and make it mandatory in v9.0.0?

**Review by tomheller — DISMISSED** [review]

File: libs/barista-components/chart/src/selection-area/selection-area.ts (hunk 1 above)
Comment (ffriedl89) [reply libs/barista-components/chart/src/selection-area/selection-area.ts:197]: I've added the comment here as well, thx for noticing!

**Comment by ffriedl89** [comment]
cherry-picked manually

**Pull request merged**